source/Mlos.Python/mlos/Optimizers/OptimizationProblem.py

```diff
@@ -99,19 +104,12 @@ def __init__(
 
         # Fit functions / surrogate models will be fed features consisting of both context and parameters.
         # Thus, the feature space is comprised of both context and parameters.
-        self.feature_space = SimpleHypergrid(
-            name="features",
-            dimensions=[
-                CategoricalDimension(name="contains_parameters", values=[True]),
-                CategoricalDimension(name="contains_context", values=[self.context_space is not None])
-            ]
-        ).join(
-            subgrid=self.parameter_space,
-            on_external_dimension=CategoricalDimension(name="contains_parameters", values=[True])
-        ).join(
-            subgrid=self.context_space,
-            on_external_dimension=CategoricalDimension(name="contains_context", values=[True])
-        )
+        parameter_names = {parameter.name for parameter in parameter_space.root_dimensions}
+        context_names = [context.name for context in context_space.root_dimensions]

```

Comment:
@amueller the error you're running into comes from the fact that currently the c# side doesn't provide any context space in the OptimizationProblem setup during the CreateOptimizer request.

I think some error handling here to default to an empty context space (the current method?) should solve that for now.

There's discussion from @byte-sculptor and myself on whether to require context or not already.

Reply:
I think for now we should have it emit a warning and continue with empty context.
Eventually, our goal is I think to automatically capture some context features (e.g. OS versions/HW specs).
We can add that in a future PR or I can try and help submit a rudimentary patch for the C# side of things to this PR if you want (just for proof of concept purposes).